app/Platform/Services/AwardEarnersService.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Services;
+
+use App\Models\PlayerBadge;
+use Illuminate\Database\Eloquent\Builder;
+
+class AwardEarnersService
+{
+    private int $awardType;
+    private int $awardId;
+    private int $awardExtra;
+
+    public function initialize(int $type, int $id, int $extra): void
+    {
+        $this->awardType = $type;
+        $this->awardId = $id;
+        $this->awardExtra = $extra;
+    }
+
+    /**
+     * @return Builder<PlayerBadge>
+     */
+    private function baseQuery(): Builder
+    {
+        return PlayerBadge::query()
+            ->where('AwardType', $this->awardType)
+            ->where('AwardData', $this->awardId)
+            ->where('AwardDataExtra', $this->awardExtra)
+            ->whereHas('user', function ($query) {
+                return $query->tracked();
+            });
```

Comment:
I know `unranked_users` was implemented after this was written. Apologies for my delayed review on this PR. This filter may be a good candidate for using that table.